### ISSUE TITLE
Add keyword: Attach to Running Browser (Extend `Connect To Browser`)

### ISSUE THREAD

**Issue by mardukbp** [MarketSquare/robotframework-browser#2676]
**Is your feature request related to a problem? Please describe.**
Currently, in order to try Browser Library keywords in a given page
one has to execute a test using a debugger to pause the execution 
upon visiting the page and then call the keywords. This is slow and 
cumbersome.

**Describe the solution you'd like**
Playwright provides the API call [BrowserType.connectOverCDP](https://playwright.dev/docs/api/class-browsertype#browser-type-connect-over-cdp) to attach 
to an existing browser instance using the Chrome DevTools Protocol. 

Implementing this functionality in the keyword Attach to Running Browser 
with the debugger address as parameter would allow to test Browser Library
keywords with a currently open browser. Using for example the interactive 
Robot Framework REPL [irobot](https://github.com/imbus/robotframework-debug) and its future VS Code Notebook incarnation.

**Describe alternatives you've considered**
Robocorp's RPA Framework provides the Selenium-based keyword [Attach Chrome Browser](https://robocorp.com/docs/libraries/rpa-framework/rpa-browser-selenium/keywords#attach-chrome-browser). 
But I am aware of the many advantages of Browser Library :)

**Additional context**
See [[Question] Attaching playwright to an existing browser window?](https://github.com/microsoft/playwright/issues/1985)

Of course this keyword would currently only work with Chromium-based browsers. 
This is enough for the purposes of experimenting with Browser Library keywords.
And maybe one day it will be possible to implement it for Firefox and WebKit too.

**Comment by aaltat** [comment]
This sounds good addition to https://marketsquare.github.io/robotframework-browser/Browser.html#Connect%20To%20Browser keyword and when quickly looking at the Playwright API, it is not complicated to implement. Most likely writing an acceptance test is hardest part. Are you able to provide a PR? 

@allcontributors please add  @mardukbp for ideas

**Comment by Snooz82** [comment]
Also test `Connect To Browser` if this is handling existing contexts and pages. If not, this shall be fixed as well.

See:
```typescript
const browser = await playwright.chromium.connectOverCDP('http://localhost:9222');
const defaultContext = browser.contexts()[0];
const page = defaultContext.pages()[0];
```

**Comment by mardukbp** [comment]
`Connect to Browser` works with a browser that was started as shown below:

```javascript
const { chromium } = require('playwright');  // Or 'webkit' or 'firefox'.

(async () => {
  const browserServer = await chromium.launchServer({headless: false});
  const wsEndpoint = browserServer.wsEndpoint();
  console.log(wsEndpoint);
})();
```

The [BrowserServer](https://playwright.dev/docs/api/class-browserserver) does not seem to be stateful though.

**Comment by Snooz82** [comment]
@mardukbp yes yes.
but it only seems to create a browser in `RF Browser`'s Browser Catalog and no context and no page.
Even though they may be open. So that should be fixed as well.

**Comment by Snooz82** [comment]
I am working on that now.

First step to make all this connection to existing Browser easier was to implement #3264 

This gives the possibility to run a browser from cli on one computer and connect to it from another.

The `Connect To Browser` keyword will be then extended to also be able to connect to a CDP server via http or websocket.
There are howevery some internals that has to be fixed. For example if a Browser already has Contexts and Pages oben, these should be added to the Browser Catalog, to be also accessible via Browser lib.

**Comment by Snooz82** [comment]
Connect To Browser now has an additional argument: `use_cdp` 
If `use_cdp` is True it uses Chrome DevTool Protocol instead of Playwright websockets to connect.
Implemented in #3263
